details/operator-sdk/controllers/astraconnector_controller.go

```diff
@@ -468,3 +479,52 @@ func (r *AstraConnectorController) needsReconcile(connector v1.AstraConnector) b
 	}
 	return false
 }
+
+// waitForStatusUpdate waits for the status of an AstraConnector to be updated in Kubernetes.
+// Status().Update() function returns even though the update is still in progress on k8s.
+// Polling to make sure this function exits only after status subresource update is reflected in k8s.
+func (r *AstraConnectorController) waitForStatusUpdate(astraConnector *v1.AstraConnector, log logr.Logger) error {
+	interval := 2 * time.Second
+	timeout := 2 * time.Minute
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	// Start polling process
+	err := errors.Wrap(wait.PollUntilContextTimeout(ctx, interval, timeout, true,
+		func(ctx context.Context) (bool, error) {
+			current := astraConnector.DeepCopy()
+			err := r.Get(ctx, types.NamespacedName{Name: astraConnector.Name, Namespace: astraConnector.Namespace}, current)
+			if err != nil {
+				log.Error(err, "Failed to get the current status of the AstraConnector. Retrying...")
+				return false, nil
+			}
+
+			astraConnectorStatusJson, err := json.Marshal(astraConnector.Status)
+			if err != nil {
+				log.Error(err, "Failed to marshal astraConnector.Status to JSON")
+				return false, nil
+			}
+
+			currentStatusJson, err := json.Marshal(current.Status)
+			if err != nil {
+				log.Error(err, "Failed to marshal current.Status to JSON")
+				return false, nil
+			}
+
+			// If the status has not been updated yet, log the current and expected statuses and continue polling.
+			if string(astraConnectorStatusJson) != string(currentStatusJson) {
+				log.Info("AstraControlCenter instance status subresource update is in progress... retrying",
+					"Expected status", astraConnector.Status, "Actual status", current.Status)
+				return false, nil
+			}
+
+			// Otherwise stop polling
+			return true, nil
+		}), fmt.Sprintf("AstraConnector status is not updated even after %s", timeout))
+
+	if err == nil {
+		log.Info("Updated AstraConnector status successfully")
```

Comment:
This function doesn't actually update the status so probably should say Status update was reflected in k8s or something? In case this gets used elsewhere that doesn't immediately follow an update call?